py/objmodule.c

```diff
@@ -61,7 +61,20 @@ STATIC void module_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
         mp_map_elem_t *elem = mp_map_lookup(&self->globals->map, MP_OBJ_NEW_QSTR(attr), MP_MAP_LOOKUP);
         if (elem != NULL) {
             dest[0] = elem->value;
+        #if MICROPY_PEP562
+        } else {
+            if (attr != MP_QSTR___getattr__) {
+                mp_obj_t dest2[3];
+                mp_load_method_maybe(self_in, MP_QSTR___getattr__, dest2);
```

Comment:
This could be optimised to use the existing `dest` array instead of a new `dest2`.  We know that if the call succeeds it will be a function, not a method, so only need 2 entries in dest.

Reply:
Furthermore, it could use `mp_map_lookup(&self->globals->map, MP_OBJ_NEW_QSTR(MP_QSTR___getattr__), MP_MAP_LOOKUP)` directly.